kinescope/downloader.py

```diff
@@ -67,24 +67,24 @@ def _decrypt_video(self, source_filepath: str | PathLike,
 
     def _get_license_key(self) -> str:
         try:
-            return b64decode(
-                self.http.post(
-                    url=self.kinescope_video.get_clearkey_license_url(),
-                    headers={'origin': KINESCOPE_BASE_URL},
-                    json={
-                        'kids': [
-                            b64encode(bytes.fromhex(
-                                self.mpd_master
-                                .periods[0]
-                                .adaptation_sets[0]
-                                .content_protections[0]
-                                .cenc_default_kid.replace('-', '')
-                            )).decode().replace('=', '')
-                        ],
-                        'type': 'temporary'
-                    }
-                ).json()['keys'][0]['k'] + '=='
-            ).hex() if self.mpd_master.periods[0].adaptation_sets[0].content_protections else None
+            with self.http.post(
+                url=self.kinescope_video.get_clearkey_license_url(),
+                headers={'origin': KINESCOPE_BASE_URL},
+                json={
+                    'kids': [
+                        b64encode(bytes.fromhex(
+                            self.mpd_master
+                            .periods[0]
+                            .adaptation_sets[0]
+                            .content_protections[0]
+                            .cenc_default_kid.replace('-', '')
+                        )).decode().replace('=', '')
+                    ],
+                    'type': 'temporary'
+                }
+            ) as post:
+                return b64decode(post.json()['keys'][0]['k'] + '=='
+                ).hex() if self.mpd_master.periods[0].adaptation_sets[0].content_protections else None
```

Comment:
Looks pretty complicated, what do you think about splitting this inline into parts?